### PR TITLE
IOS-11089 Migrate all UIAlertViews to UIAlertControllers

### PR DESCRIPTION
**Pull request by jdizoglio**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jdizoglio has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m [hunk 1]
```diff
@@ -242,13 +242,28 @@ - (void)failAuthenticationWithConnection:(NSURLConnection *)connection
...
+                                                                                 message:message
+                                                                          preferredStyle:UIAlertControllerStyleAlert];
+        
+        __weak typeof(self) weakSelf = self;
+        UIAlertAction *okAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"OK", @"Label: Allow the user to accept the current condition, often used on buttons to dismiss alerts")
+                                                           style:UIAlertActionStyleDefault
+                                                         handler:^(UIAlertAction *action) {
+                                                             [alertController dismissViewControllerAnimated:YES completion:nil];
+                                                             
+                                                             if (self.completionBlock) {
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m]: Should we put this stuff in the completion of the dismissal?

Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m]: Also, this should be weakSelf

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m [hunk 2]
```diff
@@ -506,17 +521,51 @@ - (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticatio
...
+                        [alertController addAction:cancelAction];
+                        [alertController addAction:submitAction];
+                        
+                        [alertController addTextFieldWithConfigurationHandler:^(UITextField *textField)
+                         {
+                             textField.placeholder = NSLocalizedString(@"Username", @"Alert view text placeholder: Placeholder for where to enter user name for SSO authentication challenge");
+                         }];
+                        
+                        [alertController addTextFieldWithConfigurationHandler:^(UITextField *textField) {
+                             textField.secureTextEntry = YES;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m]: Just to double check - What's the default placeholder on this textField?

**Comment by boxtcarpel** [comment]
+2 LGTM

:bear:

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m (hunk 1 above)
Comment (jdizoglio) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m]: done

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m (hunk 2 above)
Comment (jdizoglio) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m]: the new controller does not have a default password placeholder text. I updated this to set the password placeholder text. To note, this is in the english files, but is not translated.

**Pull request merged**